editoast/src/views/projection.rs

```diff
@@ -356,6 +409,149 @@ pub async fn compute_projected_train_paths(
     Ok(project_path_result)
 }
 
+#[derive(Clone, Debug, Educe)]
+#[educe(Default)]
+pub struct OpRefAndTime {
+    /// Arrival time of the train at this OP
+    time: u64,
+    #[educe(Default(expression = OperationalPointIdentifier::OperationalPointId { operational_point: Identifier::default() }))]
+    op_ref: OperationalPointIdentifier,
+}
+
+#[derive(Debug, Default)]
+pub struct TrainToProjectOnOP {
+    space_time_curve: Option<SpaceTimeCurve>,
+    ops: Vec<OpRefAndTime>,
+}
+
+impl TrainToProjectOnOP {
+    pub fn from_ts_and_sim(ts: TrainSchedule, sim: Arc<simulation::Response>) -> Self {
```

Comment:
I believe this function should consume references.


```suggestion
    pub fn from_ts_and_sim(ts: &TrainSchedule, sim: &simulation::Response) -> Self {
```

But because the caller might only have `Arc`, it might need something like this


```suggestion
    pub fn from_ts_and_sim(ts: impl AsRef<TrainSchedule>, sim: impl AsRef<simulation::Response>) -> Self {
```

Reply:
Fixed with @leovalais ' comments